File: app/src/main/java/crabster/rudakov/sberschoollesson30hwk/domain/interactors/IDictionaryInteractor.java
package crabster.rudakov.sberschoollesson30hwk.domain.interactors;

import androidx.annotation.*;

import java.util.*;

import crabster.rudakov.sberschoollesson30hwk.domain.models.*;
import io.reactivex.*;

public interface IDictionaryInteractor {

    Completable add(@NonNull DictionaryItem dictionaryItem);

    Single<List<DictionaryItem>> getList();

    Completable delete(long id);

}

File: app/src/main/java/crabster/rudakov/sberschoollesson30hwk/domain/interactors/IDictionaryInteractor.kt
package crabster.rudakov.sberschoollesson30hwk.domain.interactors

import crabster.rudakov.sberschoollesson30hwk.domain.models.DictionaryItem
import crabster.rudakov.sberschoollesson30hwk.domain.repositories.models.DictionaryItemModel
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Single

interface IDictionaryInteractor {

    fun add(dictionaryItem: DictionaryItem): Completable

    fun getList(): Single<List<DictionaryItemModel>>

    fun delete(id: Long): Completable

}